velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class DecimalArithmeticTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testDecimalExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    VectorPtr result =
+        evaluate<SimpleVector<T>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+    ASSERT_EQ(result->type()->toString(), expected->type()->toString());
```

Comment:
This is not necessary. Should be part of assertEqualVectors.

Reply:
This is required to make sure precision and scales match and moving it inside assertEqualVectors failed other Tests. Refer: https://app.circleci.com/pipelines/github/facebookincubator/velox/10148/workflows/b3be71c1-2d5a-46ac-a91b-209547f347c3/jobs/58400.

So retaining it here.

Reply:
We now have an `equivalent` method for type. We can use that inside assertEqualsVector instead of matching typeKind.

Reply:
Let's do that.

Reply:
@karteekmurthys can you address this comment?